conans/test/integration/build_requires/test_install_test_build_require.py

```diff
@@ -0,0 +1,123 @@
+                 "openssl/conanfile.py": openssl})
+
+    client.run("create tool tool/1.0@")
+    client.run("create openssl openssl/1.0@")
+    client.run("create cmake mycmake/1.0@")
+    return client
+
+
+def test_build_require_test_package(client):
+    test_cmake = textwrap.dedent(r"""
+        import os, platform
+        from conans import ConanFile
+        from conans.tools import save, chdir
+        class Pkg(ConanFile):
+            settings = "os"
+            generators = "cmake_find_package"  # No find_package should be generated!
+            test_build_require = True
+            {}
+
+            def build(self):
+                mybuild_cmd = "mycmake.bat" if platform.system() == "Windows" else "mycmake.sh"
+                self.run(mybuild_cmd)
+
+            def test(self):
+                pass
+        """)
+
+    # Test with extra build_requires to check it doesn't interfere or get deleted
+    for br in ("", 'build_requires="tool/1.0"', 'build_requires=("tool/1.0", )',
+               'build_requires=["tool/1.0"]'):
+        client.save({"cmake/test_package/conanfile.py": test_cmake.format(br)})
+        # This works when using the build-host contexts
+        client.run("create cmake mycmake/1.0@ -pr:b=default --build=missing")
+        if "tool" in br:
+            assert "mycmake/1.0 (test package): Applying build-requirement: tool/1.0" in client.out
+
+        assert "mycmake/1.0 (test package): Applying build-requirement: openssl/1.0" in client.out
+        assert "mycmake/1.0 (test package): Applying build-requirement: mycmake/1.0" in client.out
+        assert "cmake_find_package" not in client.out
+
+        system = {"Darwin": "Macos"}.get(platform.system(), platform.system())
+        assert "MYCMAKE={}!!".format(system) in client.out
+        assert "MYOPENSSL={}!!".format(system) in client.out
+
+
+def test_build_require_conanfile_text(client):
+    client.save({"conanfile.txt": "[build_requires]\nmycmake/1.0"}, clean_first=True)
+    client.run("install . -g virtualenv")
+    cmd = ". ./activate.sh && mycmake.sh" if platform.system() != "Windows" else \
+        "activate.bat && mycmake.bat"
+    client.run_command(cmd)
+    system = {"Darwin": "Macos"}.get(platform.system(), platform.system())
+    assert "MYCMAKE={}!!".format(system) in client.out
+    assert "MYOPENSSL={}!!".format(system) in client.out
+
+
+def test_build_require_command_line_no_context(client):
+    client.run("install mycmake/1.0@  -g virtualenv")
+    cmd = ". ./activate.sh && mycmake.sh" if platform.system() != "Windows" else \
+        "activate.bat && mycmake.bat"
+    client.run_command(cmd)
+    system = {"Darwin": "Macos"}.get(platform.system(), platform.system())
+    assert "MYCMAKE={}!!".format(system) in client.out
+    assert "MYOPENSSL={}!!".format(system) in client.out
+
+
+def test_build_require_command_line_build_context(client):
+    client.run("install mycmake/1.0@ --build-require -g virtualenv -pr:b=default")
```

Comment:
Shouldn't we change the build profile to see if the output changes?

Reply:
Not sure what you mean, the build-profile is different in each tested platform, and the output changes accordingly, it is parameterized to the system